### DISCUSSION TITLE
[Polkadot] terminating with uncaught exception of type std::invalid_argument: Invalid address string

### DISCUSSION THREAD

**Discussion by Raden-Hor** [trustwallet/wallet-core#2002] [Q&A]
After I change network to Westend it crash app during sign transaction
`func signDot(){
        let address = wallet.getAddressForCoin(coin: .polkadot)
        let genesisHash = Data(hexString: "0x91b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3")!
        let pk = wallet.getKeyForCoin(coin: .polkadot)
        print(address)
        let input = PolkadotSigningInput.with {
            $0.genesisHash = genesisHash
            $0.blockHash = Data(hexString: "0x1ee93bd05b18ab7f7f7b27abc040bb3a92bab3ddbf263b4b1cefcad89775b7d3")!
            $0.nonce = 1
            $0.specVersion = 9150
            $0.network = .UNRECOGNIZED(42) // Westend
            $0.transactionVersion = 6
            $0.privateKey = pk.data
            $0.era = PolkadotEra.with {
                $0.blockNumber = 9389374
                $0.period = 64
            }
            $0.balanceCall.transfer = PolkadotBalance.Transfer.with {
                $0.toAddress = "16H37bqdrLsq14YjAJcwG2RhSm89uuFLgPXWtKNdBP5V3dYi"
                $0.value = Data(hexString: "0x02540be400")! // 1 DOT
            }
        }
        let output: PolkadotSigningOutput = AnySigner.sign(input: input, coin: .polkadot)
        print("====> \(output.encoded.hexString)")
    }`

**Comment by gituser** [comment]
@Raden-Hor it won't work for westend, you need to patch wallet-core first.

**Reply by Raden-Hor** [reply]
Thank u @gituser
